
---
title: "Genetic correlation"
author: "by [Yan Holtz](https://github.com/holtzy/) - `r format(Sys.time(), '%d %B %Y')`"
---


```{r echo=FALSE}
# Just to add space between the introduction of the document
knitr::asis_output(htmltools::htmlPreserve("<br><br>"))
```


This document aims to recover as much data as possible concerning genetic correlation between traits in the UKB. Several source of information are available:

* the [LDHub website](http://ldsc.broadinstitute.org) provides GC between many pairs of traits
* the [Neale lab](http://www.nealelab.is) provide GWAS summary statistics for 600+ diseases from the UKB. Using LDSC, I computed the GC between each trait. Result file is available [here](https://github.com/holtzy/UKB-Comorbidity/blob/master/0_DATA/Genetic_Correlation_of_UKB_diseases.txt.gz).

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(DT)

# Load ICD
load("0_DATA/WHO_disease_classification_clean.R")
```









#GC from LDHub {.tabset .tabset-fade .tabset-pills}
***
The LD Hub [website](http://ldsc.broadinstitute.org/lookup/) gives the genetic correlations (GC) between the pairs of many diseases. I downloaded the results for 221 traits (without 7 traits from ENIGMA). The website is [here](http://ldsc.broadinstitute.org/lookup/) and the link I clicked is [that one](http://ldsc.broadinstitute.org/static/media/LD-Hub_genetic_correlation_221x221_no_ENIGMA.xlsx). This has been done the 19/10/2017.

The problem is that it is really hard to make the link between these traits and ICD10 classification codes.

##Get data
Load the file, reformat it, make a long format, look the 100 first lines:
```{r, warning=FALSE}
LDHubGC = read.table("0_DATA/LD-Hub_genetic_correlation_221x221_no_ENIGMA.csv.gz", header = TRUE, na.strings = "/", sep=";", dec=",")

# Change col and rownames
colnames(LDHubGC) = colnames(LDHubGC) %>% gsub(".sumstats.gz", "", .) %>% gsub(".txt", "", .) %>% gsub(".gz", "", .) %>% gsub(".tab", "", .) %>% gsub(".sumstats", "", .) %>% gsub(".csv", "", .)
LDHubGC[,1] = LDHubGC[,1] %>% gsub(".sumstats.gz", "", .) %>% gsub(".txt", "", .) %>% gsub(".gz", "", .) %>% gsub(".tab", "", .) %>% gsub(".sumstats", "", .) %>% gsub(".csv", "", .)
#summary(LDHubGC)

# A few values are not understood by R, I put NA instead. ("NA", e10-8...)
# Long format
LDHubGC_long = LDHubGC %>% gather(key="disease2", value="GC" , -1) %>% rename_(disease1 = names(.)[1]) %>% mutate(GC = as.numeric(as.character(GC))) %>% filter(!is.na(GC))

# Use the first column as a rownames for the wide format
rownames(LDHubGC)=LDHubGC[,1]
LDHubGC=LDHubGC[, -1]
LDHubGC = apply(LDHubGC, 2, function(x){as.numeric(as.character(x))} )
LDHubGC=as.matrix(LDHubGC)

# show table
datatable(head(LDHubGC_long,100), rownames = FALSE, options = list(pageLength = 5) )
```


##GC distribution
First we can check what is the the distribution of GCs amoung the `r nrow(LDHubGC_long)` available correlations. It looks like we have a normal distribution of GC, centered on O.  
Three strange events however:  
  
- A pic of GC=1, bug in the dataset?  
- Some GC over 1, is it possible to have such GCs?  
- It looks like there are more GC>0 than GC<0  

```{r}
LDHubGC_long %>% 
  filter( disease1 != disease2) %>%
  filter( GC != "NA") %>%
  arrange(desc(GC)) %>%
  ggplot( aes(x=GC)) +
    geom_histogram(bins=300)

```

##Highest GC
Can we have a look at the 5 biggest GCs for curiosity?
```{r}
LDHubGC_long %>% 
  filter( disease1 != disease2) %>%
  filter( GC != "NA") %>%
  arrange(desc(GC)) %>%
  head(15) %>%
  ggplot( aes(y=GC, x=paste(disease1, disease2, sep=" - "))) +
    geom_bar(stat="identity", fill="skyblue") +
    xlab("Pair of diseases") +
    coord_flip()
```

##Heatmap
Completely useless heatmap
```{r, fig.width=9, fig.height=9}
m=LDHubGC
m[upper.tri(m)] <- t(m)[upper.tri(m)]
#m=na.omit(m)
diag(m)=NA
heatmap(m)
```












#GC from GWAS Sumstat (UKB) {.tabset .tabset-fade .tabset-pills}
***
The [Neale lab](http://www.nealelab.is) provide GWAS summary statistics for 600+ diseases from the UKB. Using LDSC, I computed the GC between each trait. Result file is available [here](https://github.com/holtzy/UKB-Comorbidity/blob/master/0_DATA/Genetic_Correlation_of_UKB_diseases.txt.gz). This part gives the code I use to compute these GC:


##GWAS Sumstat
Summary statistics of GWAS on UKB are available for many traits. There are here:

> /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results

It appears that GWAS summary statistics are available for several mental disorders:
F10,F20,F23,F25,F30,F31,F32,F33,F41,F43,F45,F52,F60,F99.



##Install LDSC
```{r, eval=FALSE, echo=TRUE}
# good repo
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION

# Install ANACONDA to have python and every libraries

# clone LDSC (the python program)
git clone https://github.com/bulik/ldsc.git

# Now I am ready to use this program. A proof:
/clusterdata/apps/anaconda/bin/python ldsc.py -h

# I need 2 files provided by LDSC: I stock them in the 
# - eur_w_ld_chr : gives the LDs
# - w_hm3.snplist : list of HapMap3 SNPs with their alleles
```



##Reformat
```{r, eval=FALSE}
# First: copy the UKB sumstat = input data
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT
cp /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results/F* .

# Then I reformat files. This takes place in the GWAS_SUMMARY directory
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT
nano script_reformat
----
for i in /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results/[A-Z][0-9]*gz ; do
  a=$(echo $i | sed 's/.*results\///')
  zcat $i | tr '\t' ' ' | sed 's/:/ /g' | awk '{ print $5,$3,$4,$9,$12}' | sed 's/ytx.*/SNP A1 A2 Z p/' > /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/${a}.clean 
done
---
chmod 777 script_reformat.sh
qsub -b y -cwd -e tmp -o tmp -l vf=1G,h_vmem=1G "/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_reformat.sh" 
```


##Mudge

```{r, eval=FALSE}
# ----- On a specific file
i=/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A04.assoc.tsv.gz.clean
a=$(echo $i | sed 's/.assoc.*//' | sed 's/.*SUMSTAT\///')
/clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/munge_sumstats.py \
    --sumstats $i \
    --N 10000 \
    --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/$a \
    --merge-alleles /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/w_hm3.snplist



# ------ One every files

nano script_mudge.sh
---
#!/bin/bash
for i in /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/*.clean; do
  a=$(echo $i | sed 's/.assoc.*//' | sed 's/.*SUMSTAT\///')
  echo $a
  /clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/munge_sumstats.py \
    --sumstats $i \
    --N 10000 \
    --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/$a \
    --merge-alleles /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/w_hm3.snplist
done
---
chmod 777 script_mudge.sh
qsub -b y -cwd  -e tmp -o tmp -l vf=30G,h_vmem=30G "/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_mudge.sh" 
```



##Compute GC
```{r, eval=FALSE}
# ----- On a specific file
# We have error for diseases F23-F25-F41-F52-F99. Several possibilities exist: heritability too low OR alleles with too strong effect.
# See https://github.com/bulik/ldsc/issues/54
/clusterdata/apps/anaconda/bin/python   /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/ldsc.py \
        --rg /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A04.sumstats.gz,/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A08.sumstats.gz \
        --ref-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --w-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/tmp_test_cor
        

# ------ On every files
nano script_GC_computation
--------
all=$(ls /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/*sumstats*)
ntot=$(echo $all | awk '{print NF}')
i=$1
for j in $(seq 1 $ntot); do
    if [ "$j" -gt "$i" ]; then
      
      # Get the 2 files that I am going to study
      a=$( echo $all | awk -v n=$i '{print $n}' )
      b=$( echo $all | awk -v n=$j '{print $n}' )
      
      # Make  clean names
      namea=$( echo $a | sed 's/.sumstats.gz//' | sed 's/.*GWAS_SUMSTAT\///')
      nameb=$( echo $b | sed 's/.sumstats.gz//' | sed 's/.*GWAS_SUMSTAT\///')
      echo $namea, $nameb
      
      # run the program
      /clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/ldsc.py \
        --rg ${a},${b} \
        --ref-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --w-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/Cor_${namea}_${nameb}

    fi
done
--------
chmod 777 script_GC_computation
# Test on a number
qsub -b y -cwd -N GC_computation -e test.e -o test.o -l vf=30G,h_vmem=30G "/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_GC_computation 640"

# Send it on every number
for i in $(seq 1 600) ; do
  qsub -b y -cwd -N GC_computation -e GC.e -o GC.o -l vf=20G,h_vmem=20G "/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_GC_computation $i"
done
```





##Summarize
For each pair, I need to get the result of LDSC and write them in a clean file.
```{r, eval=FALSE}
---
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GENETIC_CORRELATION/
echo "p1 p2 rg se z p h2_obs h2_obs_se h2_int h2_int_se gcov_int gcov_int_se"> /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt
for i in /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/Cor_* ; do
  val=$(cat $i | grep '^/ib.*sumstat' | cut -d" " -f4-)
  d1=$( echo $i | sed 's/.*Cor_//' | sed 's/_.*//')
  d2=$( echo $i | sed 's/.*Cor.*_//' | sed 's/.log//')
  echo $d1 $d2 $val >> /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt
done 
---
nano script_summarize_GC.sh
chmod 777 script_summarize_GC.sh
qsub -b y -cwd -N tmp_summarize_GC -e tmp.e -o tmp.o -l vf=20G,h_vmem=20G "/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/script_summarize_GC.sh"

# How many disease are remaining (for some of them I cannot calculate GC cause heritability to low)
more Genetic_Correlation_of_UKB_diseases.txt | grep -v "NA" | grep "^A04" | wc -l
--> 380 diseases remaining, 72000 pairs

# Remove NAs and zip
more Genetic_Correlation_of_UKB_diseases.txt | grep -v "NA" > tmp
mv tmp Genetic_Correlation_of_UKB_diseases.txt
gzip Genetic_Correlation_of_UKB_diseases.txt
```

Transfert it locally
```{r, eval=FALSE}
cd ~/Dropbox/QBI/3_UK_BIOBANK_COMO_PROJECT/1_FILES
scp  uqyholtz@cluster.qbi.uq.edu.au://ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt.gz .
```



##Heritabilities
```{r, eval=FALSE}
echo "trait,h2,h2Se,lambda,meanChi2,intercept,interceptSe" > Heritability_of_UKB_diseases
for i  in $(ls *sumstats.gz | sed 's/.sumstats.gz//' ) ; do
  echo $i
  toread=$(ls Cor_${i}_* | head -1)
  a=$(cat ${toread} | grep -A7 "Heritability of phenotype 1" | cut -d":" -f2,3 | grep -v "Herit" | sed '1d' | sed 's/^ //' | sed 's/(//' | sed 's/)//' |  head -4 | tr "\n" " " | sed 's/ /,/g' | sed 's/,$/\n/')
  echo $i,$a >> Heritability_of_UKB_diseases 
done
more Heritability_of_UKB_diseases
cat Heritability_of_UKB_diseases | grep -v ",$" > tmp
mv tmp Heritability_of_UKB_diseases
```

This heritabilities are wrong compare to what is provided on the Neale Lab website. So it'is better to load their heritabilities.
```{r}
Herit=read.table("0_DATA/ukbb_all_h2part_results_small.txt", sep="\t", header = T)
Herit=Herit[ substr(Herit$phenotype, 1, 1) %in% LETTERS , c(1:20) ]
Herit=Herit %>% arrange(phenotype)
datatable(Herit, rownames = FALSE, options = list(pageLength = 5) )
```











#GC Description {.tabset .tabset-fade .tabset-pills}
***
```{r}
# Load file
UKBGC=read.table("0_DATA/Genetic_Correlation_of_UKB_diseases.txt.gz", header=T, sep=" ")

# Number of diseases involdved?
num=length( unique( c(as.character(UKBGC$p1), as.character(UKBGC$p2)) ) )

# Number of pairs
num_pair=nrow(UKBGC)

#How many pairs are significant? (Absolute value and percentage)
nsignif=UKBGC %>% filter(p<0.05) %>% nrow()
```


I now have the genetic correlation computed for many diseases. Sometimes the heritability of a disease was to low to use it in GCs. Finally I have GCs info for **`r num`** different diseases, and **`r num_pair` pairs** of diseases. Among them, `r nsignif` (`r nsignif / nrow(UKBGC) *100` %) are significatively different from 0.


##Table
All the significant GC values are stored in this interactive table:
```{r}
signif=UKBGC %>% filter(p<0.05)
datatable(signif, rownames = FALSE, options = list(pageLength = 5) )
```



##Distribution
```{r, warning=FALSE, message=FALSE}
UKBGC %>%
  ggplot( aes(x=rg)) +
    geom_histogram(bins=1000)
```


##Sullivan comparison
Can I trust my values? Can I compare with something well known? It looks to be more or less correct. For example, the paper "An Atlas of Genetic Correlations across Human Diseases and Traits" by Bulik-Sullivan et al. gives strong rg for Bipolar disBipolar Disorder (F31) with Major Depression (F32, F33) and with Schizophrenia (F20). Let's check the value of this diseases in my calculations:
```{r}
# Bipolar Disorder (F31) with Major Depression (F32, F33) with Schizophrenia (F20)
to_keep=c("F20","F31","F32","F33")
UKBGC %>% filter(p1 %in% to_keep)  %>% filter(p2 %in% to_keep)
```




##Extreme values
What are the extreme values? These extreme values have really high standard deviation. They are not even significatively different from 0!
```{r, warning=FALSE, message=FALSE}
tmp=UKBGC %>% 
  arrange(desc(abs(rg))) %>% 
  select(p1, p2, rg, se, z, p) %>%
  head(100) %>%
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p1"="coding_L3")) %>% 
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p2"="coding_L3")) %>% 
  unique() %>%
  head()
datatable(tmp, rownames = FALSE, options = list(pageLength = 5) )
```

Let's check the extreme AND significant genetic correlation. It is completely different, the highest significant GC is -1.7269.
```{r}
tmp=UKBGC %>% 
  arrange(desc(abs(rg))) %>% 
  filter(p<0.05) %>%
  select(p1, p2, rg, se, z, p) %>%
  head(100) %>%
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p1"="coding_L3")) %>% 
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p2"="coding_L3")) %>% 
  unique() %>%
  head()
datatable(tmp, rownames = FALSE, options = list(pageLength = 5) )
```


Let's check the lowest pvalues?
```{r}
tmp=UKBGC %>% 
  arrange(p) %>% 
  filter(p<0.05) %>%
  select(p1, p2, rg, se, z, p) %>%
  head(100) %>%
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p1"="coding_L3")) %>% 
  left_join(. , ICD[,c("coding_L3", "meaning_L3")], by=c("p2"="coding_L3")) %>% 
  unique() %>%
  head() 
```



##Heatmap
Can I check a heatmap of this GC for mental disease??
```{r}
UKBGC %>%
  filter(substr(p1, 1, 1)=="F") %>%
  filter(substr(p2, 1, 1)=="F") %>%
  ggplot( aes(x=p1, y=p2, fill=abs(rg))) +
    geom_tile(color="orange")
```

A heatmap for all the disease that have at least one strong correlation?
```{r, fig.height=10, fig.width=10}
tmp=UKBGC %>%
  filter( p<0.01) %>%
  select(p1,p2)
tokeep=unique(c(as.character(tmp$p1), as.character(tmp$p2)))

UKBGC %>%
  filter(p1 %in% tokeep) %>%
  filter(p2 %in% tokeep) %>%
  ggplot( aes(x=p1, y=p2, fill=rg)) +
    geom_tile(color="orange")
```



##Per chapter
Can I check the global GC into each disease chapter
```{r, warning=FALSE}
UKBGC %>%
  mutate(letter1=substr(p1, 1, 1)) %>%
  mutate(letter2=substr(p2, 1, 1)) %>%
  filter(letter1==letter2) %>%
  mutate(rg=abs(rg)) %>%
  ggplot( aes(x=reorder(letter1, rg, FUN = median), y=rg, fill=letter1)) +
    geom_boxplot(varwidth = TRUE) +
    theme(legend.position="none") +
    xlab("Disease group") +
    ylab("Genetic correlation") +
    ylim(0,4)
```


Same stuff, but only with significant rg. Would be better to do it with prevelence filter instead.
```{r, warning=FALSE}
UKBGC %>%
  mutate(letter1=substr(p1, 1, 1)) %>%
  mutate(letter2=substr(p2, 1, 1)) %>%
  filter(letter1==letter2) %>%
  mutate(rg=abs(rg)) %>%
  filter(p<0.05) %>%
  ggplot( aes(x=reorder(letter1, rg, FUN = median), y=rg, fill=letter1)) +
    geom_boxplot(varwidth = TRUE) +
    theme(legend.position="none") +
    xlab("Disease group") +
    ylab("Genetic correlation") +
    ylim(0,4)
```

Number of significant pairs for each group of disease? There is NO significant rg for the mental disease group! This make sense since we saw before that the prevalence of mental disease is not that high compared to other group. 
```{r, warning=FALSE}
UKBGC %>%
  mutate(letter1=substr(p1, 1, 1)) %>%
  mutate(letter2=substr(p2, 1, 1)) %>%
  filter(letter1==letter2) %>%
  filter(p<0.05) %>%
  group_by(letter1) %>%
  count() %>%
  arrange(desc(n))
```
















#Heritability description {.tabset .tabset-fade .tabset-pills}

```{r, warning=FALSE}
#I add a column to check if the heritability is significatively different from 0, and I add the definition of the disease.
Herit = Herit %>% 
  mutate(signif=ifelse( h2_p < 0.05, "yes", "no")) %>% 
  arrange(desc(h2_liability)) %>% 
  left_join(. , ICD, by=c("phenotype"="coding_L3")) %>% 
  select(phenotype, N, N_case, N_control, prevelence, h2_liability, h2_liability_se, h2_p, signif, meaning_L1, meaning_L2, meaning_L3, short) %>% unique() 

nsignif=Herit %>% filter(signif=="yes") %>% nrow()
```

I now have heritabilities for `r nrow(Herit)` traits. `r nsignif`of them are significant.



##Distribution
Let's start by checking the distribution of these heritabilities? Most of them are very low. I even have a lot of value < 0. This is possible because I use LDSC to calculate these heritabilities.
```{r}
ggplot(Herit, aes(x=h2_liability)) + 
  geom_histogram(fill="skyblue", color="grey", bins=50)
```


##Most heritable
What are the 10 biggest heritability?
```{r}
tmp=Herit %>% arrange(desc(h2_liability)) %>% head(10)
datatable(tmp, rownames = FALSE, options = list(pageLength = 5) )
```

##Per group
Visualise heritability of each group
```{r, fig.height=7, fig.width=12}
Herit %>%
  arrange(meaning_L1, h2_liability) %>%
  mutate(phenotype=factor(phenotype, phenotype)) %>%
  mutate(h2_liability = ifelse(h2_liability>0, h2_liability,0 )) %>%
  filter(short != "unclassified") %>%
  ggplot( aes(x=phenotype, y=h2_liability, color=log(h2_liability))) +
    geom_point() +
    coord_flip() +
    facet_wrap(~short, scale="free_y", ncol=6) +
    theme(
      legend.position="none"
    ) +
    xlab("Heritability ('liability h2')") +
    ylab("")
```

##Link with prevalence
What is the link between Heritability and prevalence?
```{r}
Herit %>%
  ggplot( aes(y=h2_liability, x=prevelence*100)) +
    geom_point()
```





