<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="by Yan Holtz - 25 January 2018" />


<title>UKBiobank Exploratory Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comorbidity in the UK Biobank</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_DataWrangling.html">Get data</a>
</li>
<li>
  <a href="2_Exploratory_Analysis.html">explore UKB</a>
</li>
<li>
  <a href="3_Comorbidity_Analysis.html">Como</a>
</li>
<li>
  <a href="5_Compute_GeneticCorrelation">Genet Cor</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<a href="https://github.com/holtzy/" class="github-corner" aria-label="View source on Github"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#2ecc71; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">UKBiobank Exploratory Analysis</h1>
<h4 class="author"><em>by <a href="https://github.com/holtzy/">Yan Holtz</a> - 25 January 2018</em></h4>

</div>


<br><br>
<p>This document aims to recover as much data as possible concerning genetic correlation between traits. It studies different options. Sometimes it just take GC from an external resource, sometimes I compute it myself.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<div id="ldhub" class="section level1 tabset tabset-fade tabset-pills">
<h1><span class="header-section-number">1</span> LDHub</h1>
<hr />
<div id="get-data" class="section level2">
<h2><span class="header-section-number">1.1</span> Get data</h2>
<p>The LD Hub <a href="http://ldsc.broadinstitute.org/lookup/">website</a> gives the genetic correlations (GC) between the pairs of many diseases. I downloaded the results for 221 traits (without 7 traits from ENIGMA) using data from LD Hub. The website is <a href="http://ldsc.broadinstitute.org/lookup/">here</a> and the link I clicked is <a href="http://ldsc.broadinstitute.org/static/media/LD-Hub_genetic_correlation_221x221_no_ENIGMA.xlsx">that one</a>. This has been done the 19/10/2017.</p>
<p>This file has been saved in the dropbox. I can load it</p>
<pre class="r"><code>LDHubGC = read.table(&quot;0_DATA/LD-Hub_genetic_correlation_221x221_no_ENIGMA.csv.gz&quot;, header = TRUE, na.strings = &quot;/&quot;, sep=&quot;;&quot;, dec=&quot;,&quot;)</code></pre>
<p>And reformat it a little bit + Make a long format</p>
<pre class="r"><code># Change col and rownames
colnames(LDHubGC) = colnames(LDHubGC) %&gt;% gsub(&quot;.sumstats.gz&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.txt&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.gz&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.tab&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.sumstats&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.csv&quot;, &quot;&quot;, .)
LDHubGC[,1] = LDHubGC[,1] %&gt;% gsub(&quot;.sumstats.gz&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.txt&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.gz&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.tab&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.sumstats&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;.csv&quot;, &quot;&quot;, .)
#summary(LDHubGC)

# A few values are not understood by R, I put NA instead. (&quot;NA&quot;, e10-8...)
# Long format
LDHubGC_long = LDHubGC %&gt;% gather(key=&quot;disease2&quot;, value=&quot;GC&quot; , -1) %&gt;% rename_(disease1 = names(.)[1]) %&gt;% mutate(GC = as.numeric(as.character(GC)))

# Use the first column as a rownames for the wide format
rownames(LDHubGC)=LDHubGC[,1]
LDHubGC=LDHubGC[, -1]
LDHubGC = apply(LDHubGC, 2, function(x){as.numeric(as.character(x))} )
LDHubGC=as.matrix(LDHubGC)</code></pre>
</div>
<div id="gc-distribution" class="section level2">
<h2><span class="header-section-number">1.2</span> GC distribution</h2>
<p>First we can check what is the the distribution of GCs amoung the 45796 available correlations. It looks like we have a normal distribution of GC, centered on O.<br />
Three strange events however:</p>
<ul>
<li>A pic of GC=1, bug in the dataset?<br />
</li>
<li>Some GC over 1, is it possible to have such GCs?<br />
</li>
<li>It looks like there are more GC&gt;0 than GC&lt;0</li>
</ul>
<pre class="r"><code>LDHubGC_long %&gt;% 
  filter( disease1 != disease2) %&gt;%
  filter( GC != &quot;NA&quot;) %&gt;%
  arrange(desc(GC)) %&gt;%
  ggplot( aes(x=GC)) +
    geom_histogram(bins=300)</code></pre>
<p><img src="5_Compute_GeneticCorrelation_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="highest-gc" class="section level2">
<h2><span class="header-section-number">1.3</span> Highest GC</h2>
<p>Can we have a look at the 5 biggest GCs for curiosity?</p>
<pre class="r"><code>LDHubGC_long %&gt;% 
  filter( disease1 != disease2) %&gt;%
  filter( GC != &quot;NA&quot;) %&gt;%
  arrange(desc(GC)) %&gt;%
  head(15) %&gt;%
  ggplot( aes(y=GC, x=paste(disease1, disease2, sep=&quot; - &quot;))) +
    geom_bar(stat=&quot;identity&quot;, fill=&quot;skyblue&quot;) +
    xlab(&quot;Pair of diseases&quot;) +
    coord_flip()</code></pre>
<p><img src="5_Compute_GeneticCorrelation_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="heatmap" class="section level2">
<h2><span class="header-section-number">1.4</span> Heatmap</h2>
<p>Heatmap of these GC:</p>
<pre class="r"><code>m=LDHubGC
m[upper.tri(m)] &lt;- t(m)[upper.tri(m)]
#m=na.omit(m)
diag(m)=NA
heatmap(m)</code></pre>
<p><img src="5_Compute_GeneticCorrelation_files/figure-html/unnamed-chunk-7-1.png" width="864" /></p>
<p>What are the disease that could potentially interest me?: -Schizophrenia.sumstats.gz</p>
</div>
</div>
<div id="summary-statistics-from-ukb" class="section level1 tabset tabset-fade tabset-pills">
<h1><span class="header-section-number">2</span> Summary Statistics from UKB</h1>
<hr />
<p>Summary statistics of GWAS on UKB are available for many traits. There are here:</p>
<pre><code>cd /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results</code></pre>
<p>It appears that GWAS summary statistics are available for several mental disorders: F10,F20,F23,F25,F30,F31,F32,F33,F41,F43,F45,F52,F60,F99.</p>
<p>With these summary, we should be able to compute genetic correlations.</p>
<div id="install-the-ldsc-program" class="section level2">
<h2><span class="header-section-number">2.1</span> Install the LDSC program</h2>
<pre class="r"><code># good repo
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION

# Install ANACONDA to have python and every libraries

# clone LDSC (the python program)
git clone https://github.com/bulik/ldsc.git

# Now I am ready to use this program. A proof:
/clusterdata/apps/anaconda/bin/python ldsc.py -h

# I need 2 files provided by LDSC: I stock them in the 
# - eur_w_ld_chr : gives the LDs
# - w_hm3.snplist : list of HapMap3 SNPs with their alleles</code></pre>
</div>
<div id="reformat-the-files-to-be-readable-by-ldsc" class="section level2">
<h2><span class="header-section-number">2.2</span> 2.2 Reformat the files to be readable by LDSC</h2>
<pre class="r"><code># First: copy the UKB sumstat = input data
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT
cp /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results/F* .

# Then I reformat files. This takes place in the GWAS_SUMMARY directory
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT
nano script_reformat
----
for i in /afm01/Q0286/huanwei.wang/reference/UKB_neale/gwas_results/[A-Z][0-9]*gz ; do
  a=$(echo $i | sed &#39;s/.*results\///&#39;)
  zcat $i | tr &#39;\t&#39; &#39; &#39; | sed &#39;s/:/ /g&#39; | awk &#39;{ print $5,$3,$4,$9,$12}&#39; | sed &#39;s/ytx.*/SNP A1 A2 Z p/&#39; &gt; /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/${a}.clean 
done
---
chmod 777 script_reformat.sh
qsub -b y -cwd -e tmp -o tmp -l vf=1G,h_vmem=1G &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_reformat.sh&quot; </code></pre>
</div>
<div id="mudge-the-files." class="section level2">
<h2><span class="header-section-number">2.3</span> 2.3 Mudge the files.</h2>
<pre class="r"><code># ----- On a specific file
i=/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A04.assoc.tsv.gz.clean
a=$(echo $i | sed &#39;s/.assoc.*//&#39; | sed &#39;s/.*SUMSTAT\///&#39;)
/clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/munge_sumstats.py \
    --sumstats $i \
    --N 10000 \
    --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/$a \
    --merge-alleles /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/w_hm3.snplist



# ------ One every files

nano script_mudge.sh
---
#!/bin/bash
for i in /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/*.clean; do
  a=$(echo $i | sed &#39;s/.assoc.*//&#39; | sed &#39;s/.*SUMSTAT\///&#39;)
  echo $a
  /clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/munge_sumstats.py \
    --sumstats $i \
    --N 10000 \
    --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/$a \
    --merge-alleles /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/w_hm3.snplist
done
---
chmod 777 script_mudge.sh
qsub -b y -cwd  -e tmp -o tmp -l vf=30G,h_vmem=30G &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_mudge.sh&quot; </code></pre>
</div>
<div id="compute-the-genetic-correlation" class="section level2">
<h2><span class="header-section-number">2.4</span> Compute the genetic correlation</h2>
<pre class="r"><code># ----- On a specific file
# We have error for diseases F23-F25-F41-F52-F99. Several possibilities exist: heritability too low OR alleles with too strong effect.
# See https://github.com/bulik/ldsc/issues/54
/clusterdata/apps/anaconda/bin/python   /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/ldsc.py \
        --rg /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A04.sumstats.gz,/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/A08.sumstats.gz \
        --ref-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --w-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/tmp_test_cor
        

# ------ On every files
nano script_GC_computation
--------
all=$(ls /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/*sumstats*)
ntot=$(echo $all | awk &#39;{print NF}&#39;)
i=$1
for j in $(seq 1 $ntot); do
    if [ &quot;$j&quot; -gt &quot;$i&quot; ]; then
      
      # Get the 2 files that I am going to study
      a=$( echo $all | awk -v n=$i &#39;{print $n}&#39; )
      b=$( echo $all | awk -v n=$j &#39;{print $n}&#39; )
      
      # Make  clean names
      namea=$( echo $a | sed &#39;s/.sumstats.gz//&#39; | sed &#39;s/.*GWAS_SUMSTAT\///&#39;)
      nameb=$( echo $b | sed &#39;s/.sumstats.gz//&#39; | sed &#39;s/.*GWAS_SUMSTAT\///&#39;)
      echo $namea, $nameb
      
      # run the program
      /clusterdata/apps/anaconda/bin/python /ibscratch/wrayvisscher/Yan_Holtz/1_SOFT/ldsc/ldsc.py \
        --rg ${a},${b} \
        --ref-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --w-ld-chr /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/eur_w_ld_chr/ \
        --out /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/Cor_${namea}_${nameb}

    fi
done
--------
chmod 777 script_GC_computation
# Test on a number
qsub -b y -cwd -N GC_computation -e test.e -o test.o -l vf=30G,h_vmem=30G &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_GC_computation 640&quot;

# Send it on every number
for i in $(seq 1 600) ; do
  qsub -b y -cwd -N GC_computation -e GC.e -o GC.o -l vf=20G,h_vmem=20G &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/script_GC_computation $i&quot;
done</code></pre>
</div>
<div id="summarize-in-a-clean-file" class="section level2">
<h2><span class="header-section-number">2.5</span> Summarize in a clean file</h2>
<p>For each pair, I need to get the result of LDSC and write them in a clean file.</p>
<pre class="r"><code>---
cd /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GENETIC_CORRELATION/
echo &quot;p1 p2 rg se z p h2_obs h2_obs_se h2_int h2_int_se gcov_int gcov_int_se&quot;&gt; /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt
for i in /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/Cor_* ; do
  val=$(cat $i | grep &#39;^/ib.*sumstat&#39; | cut -d&quot; &quot; -f4-)
  d1=$( echo $i | sed &#39;s/.*Cor_//&#39; | sed &#39;s/_.*//&#39;)
  d2=$( echo $i | sed &#39;s/.*Cor.*_//&#39; | sed &#39;s/.log//&#39;)
  echo $d1 $d2 $val &gt;&gt; /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt
done 
---
nano script_summarize_GC.sh
chmod 777 script_summarize_GC.sh
qsub -b y -cwd -N tmp_summarize_GC -e tmp.e -o tmp.o -l vf=20G,h_vmem=20G &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/script_summarize_GC.sh&quot;

# How many disease are remaining (for some of them I cannot calculate GC cause heritability to low)
more Genetic_Correlation_of_UKB_diseases.txt | grep -v &quot;NA&quot; | grep &quot;^A04&quot; | wc -l
--&gt; 380 diseases remaining, 72000 pairs

# Remove NAs and zip
more Genetic_Correlation_of_UKB_diseases.txt | grep -v &quot;NA&quot; &gt; tmp
mv tmp Genetic_Correlation_of_UKB_diseases.txt
gzip Genetic_Correlation_of_UKB_diseases.txt</code></pre>
<p>Transfert it locally</p>
<pre class="r"><code>cd ~/Dropbox/QBI/3_UK_BIOBANK_COMO_PROJECT/1_FILES
scp  uqyholtz@cluster.qbi.uq.edu.au://ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/CORRELATION/Genetic_Correlation_of_UKB_diseases.txt.gz .</code></pre>
</div>
<div id="make-a-file-with-heritabilities" class="section level2">
<h2><span class="header-section-number">2.6</span> Make a File with heritabilities</h2>
<pre class="r"><code>echo &quot;trait,h2,h2Se,lambda,meanChi2,intercept,interceptSe&quot; &gt; Heritability_of_UKB_diseases
for i  in $(ls *sumstats.gz | sed &#39;s/.sumstats.gz//&#39; ) ; do
  echo $i
  toread=$(ls Cor_${i}_* | head -1)
  a=$(cat ${toread} | grep -A7 &quot;Heritability of phenotype 1&quot; | cut -d&quot;:&quot; -f2,3 | grep -v &quot;Herit&quot; | sed &#39;1d&#39; | sed &#39;s/^ //&#39; | sed &#39;s/(//&#39; | sed &#39;s/)//&#39; |  head -4 | tr &quot;\n&quot; &quot; &quot; | sed &#39;s/ /,/g&#39; | sed &#39;s/,$/\n/&#39;)
  echo $i,$a &gt;&gt; Heritability_of_UKB_diseases 
done
more Heritability_of_UKB_diseases
cat Heritability_of_UKB_diseases | grep -v &quot;,$&quot; &gt; tmp
mv tmp Heritability_of_UKB_diseases</code></pre>
<p>â€“&gt; This heritabilities are wrong compare to what is provided on the Neale Lab website.. And I copy this file in a special folder for further analysis. And I transfert this file locally</p>
<pre class="r"><code>cd ~/Dropbox/QBI/3_UK_BIOBANK_COMO_PROJECT/1_FILES
scp  uqyholtz@cluster.qbi.uq.edu.au://ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/GENETIC_CORRELATION/GWAS_SUMSTAT/*of_UKB_diseases* .</code></pre>
</div>
</div>
<div id="summary-statistics-of-other-study" class="section level1">
<h1><span class="header-section-number">3</span> 3- Summary Statistics of other study</h1>
<p>Summary statistics of GWAS on several published studies are available for many traits. There are here:</p>
<pre class="r"><code>/afm01/Q0286/GWA_sumstats/formatted</code></pre>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
